Source/NSLayoutConstraint.m

```diff
@@ -0,0 +1,569 @@
+{
+  NSEnumerator *en = [constraints objectEnumerator];
+  NSLayoutConstraint *c = nil;
+
+  while ((c = [en nextObject]) != nil)
+    {
+      [NSLayoutConstraint _activateConstraint: c];
+    }
+}
+
++ (void) deactivateConstraints: (NSArray *)constraints
+{
+  NSEnumerator *en = [constraints objectEnumerator];
+  NSLayoutConstraint *c = nil;
+
+  while ((c = [en nextObject]) != nil)
+    {
+      [NSLayoutConstraint _removeConstraint: c];
+    }
+}
+
+- (instancetype) initWithItem: (id)firstItem 
+                    attribute: (NSLayoutAttribute)firstAttribute 
+                    relatedBy: (NSLayoutRelation)relation 
+                       toItem: (id)secondItem
+                    attribute: (NSLayoutAttribute)secondAttribute 
+                   multiplier: (CGFloat)multiplier
+                     constant: (CGFloat)constant;
+{
+  self = [super init];
+  if (self != nil)
+    { 
+      _firstItem = firstItem;
+      _secondItem = secondItem;
+      _firstAttribute = firstAttribute;
+      _secondAttribute = secondAttribute;
+      _relation = relation;
+      _multiplier = multiplier;
+      _constant = constant;
+
+      [NSLayoutConstraint _activateConstraint: self];
+    }
+  return self;
+}
+
+// Active  
+- (BOOL) isActive
+{
+  return [activeConstraints containsObject: self];
+}
+
+- (void) setActive: (BOOL)flag
+{
+  if (flag)
+    {
+      [NSLayoutConstraint _activateConstraint: self];
+    }
+  else
+    {
+      [NSLayoutConstraint _removeConstraint: self];
+    }
+}
+
+// compare and isEqual...
+- (NSComparisonResult) compare: (NSLayoutConstraint *)constraint
+{
+  if ([self priority] < [constraint priority])
+    {
+      return NSOrderedAscending;
+    }
+  else if ([self priority] > [constraint priority])
+    {
+      return NSOrderedDescending;
+    }
+  
+  return NSOrderedSame;
+}
+
+- (BOOL) isEqual: (NSLayoutConstraint *)constraint
+{
+  BOOL result = [super isEqual: constraint];
+
+  if (result == NO)
+    {
+      result =  (_firstItem == [constraint firstItem] &&
+                 _secondItem == [constraint secondItem] &&
+                 _firstAttribute == [constraint firstAttribute] &&
+                 _secondAttribute == [constraint secondAttribute] &&
+                 _relation == [constraint relation] &&
+                 _multiplier == [constraint multiplier] &&
+                 _constant == [constraint constant] &&
+                 _priority == [constraint priority]);
+    }
+
+  return result;
+}
+
+// Items
+- (id) firstItem
+{
+  return _firstItem;
+}
+
+- (NSLayoutAttribute) firstAttribute
+{
+  return _firstAttribute;
+}
+
+- (NSLayoutRelation) relation
+{
+  return _relation;
+}
+
+- (id) secondItem
+{
+  return _secondItem;
+}
+
+- (NSLayoutAttribute) secondAttribute
+{
+  return _secondAttribute;
+}
+
+- (CGFloat) multiplier
+{
+  return _multiplier;
+}
+
+- (CGFloat) constant
+{
+  return _constant;
+}
+
+- (NSLayoutAnchor *) firstAnchor
```

Comment:
The two anchors are not used in any of the other methods. Where would they get set, how about copy, coding and equality?

Reply:
They should get set in [NSLayoutConstraint initWithCoder:]... I need to add some to the .xib to test.

Reply:
I haven't implemented this just yet.